web/src/message_fetch.ts

```diff
@@ -369,11 +451,16 @@ export function load_messages(opts, attempt = 1) {
             const backoff_scale = Math.min(2 ** attempt, 32);
             const backoff_delay_secs = ((1 + Math.random()) / 2) * backoff_scale;
             let rate_limit_delay_secs = 0;
-            if (xhr.status === 429 && xhr.responseJSON?.code === "RATE_LIMIT_HIT") {
+            const xhr_schema = z.object({
+                "retry-after": z.number(),
+                code: z.string(),
+            });
+            const parsed = xhr_schema.safeParse(xhr.responseJSON);
+            if (xhr.status === 429 && parsed.data && parsed.data.code === "RATE_LIMIT_HIT") {
```

Comment:
I think you need to call this `rate_limited_error_schema`; and you can probably put `code` into the schema as a literal value, and then just check if it's a 429 and the schema parses?